pytorch_lightning/accelerators/gpu.py

```diff
@@ -54,7 +57,7 @@ def set_nvidia_flags(local_rank: int) -> None:
         devices = os.getenv("CUDA_VISIBLE_DEVICES", all_gpu_ids)
         _log.info(f"LOCAL_RANK: {local_rank} - CUDA_VISIBLE_DEVICES: [{devices}]")
 
-    def get_device_stats(self, device: Union[str, torch.device]) -> Dict[str, Any]:
+    def get_device_stats(self, device: _DEVICE) -> dict[str, Any]:
```

Comment:
Should we have same device typing for `get_nvidia_gpu_stats` and `get_nvidia_gpu_stats`. 
If pass in a str device to `get_device_stats`, we will fail silently as `get_nvidia_gpu_stats` couldn't handle it.

Reply:
If i'm not mistaken, with the changes in this PR, all types are supported for both.

Reply:
in line 109 we are calling "gpu_id = _get_gpu_id(device.index)" if device is str or int, it will fail?

Reply:
It was changed too: https://github.com/PyTorchLightning/pytorch-lightning/pull/11333/files#diff-83cb1466a7f054dee61f11ba07f6be97615216d8eb399dd468e17fd89425ec56R112

Reply:
Oh sorry, my bad, saw it now!